File: cppfastlogging/examples/file.cpp
#include "h/cppfastlogging.hpp"

using namespace logging;

// File: file.cpp
//
// Sample library usage.
int main(void)
{
    WriterConfig configs[] = {FileWriterConfig(DEBUG,
                                               "/tmp/cfastlogging.log",
                                               1024,
                                               3,
                                               -1,
                                               -1,
                                               CCompressionMethodEnum_t::Store)};
    Logging *logging = new Logging(DEBUG, "root", configs);
    logging->trace("Trace Message");
    logging->debug("Debug Message");
    logging->info("Info Message");
    logging->success("Success Message");
    logging->warn("Warning Message");
    logging->error("Error Message");
    logging->fatal("Fatal Message");
    delete logging;
    return 0;
}

File: pyfastlogging/examples/file.py
import os

import fastlogging_rs as fl
from fastlogging_rs import (
    TRACE,
    CompressionMethodEnum,
    FileWriterConfig,
)

if __name__ == "__main__":
    pathName = (
        "C:\\temp\\pyfastlogging.log" if os.name == "nt" else "/tmp/pyfastlogging.log"
    )
    fl.add_writer(
        FileWriterConfig(TRACE, pathName, compression=CompressionMethodEnum.Deflate)
    )
    fl.trace("Trace Message")
    fl.debug("Debug Message")
    fl.info("Info Message")
    fl.success("Success Message")
    fl.warning("Warning Message")
    fl.error("Error Message")
    fl.fatal("Fatal Message")
